rmw/include/rmw/types.h

```diff
@@ -518,8 +518,21 @@ typedef struct RMW_PUBLIC_TYPE rmw_gid_s
 /// Information describing an rmw message
 typedef struct RMW_PUBLIC_TYPE rmw_message_info_s
 {
+  /// Time when the DDS sample was written in the publisher queue.
```

Comment:
Let's not include the DDS specifics here and remember that it needs to be flexible in case we're not using DDS. Maybe:

```suggestion
  /// Time when the message was published by the publisher.
  /**
   * The exact point at which the timestamp is taken is not specified, but
   * it should be taken consistently at the same point in the
   * publishing process each time.
   */
```

Reply:
Yeah sounds good.
I thought mentioning the queue wasn't a bad idea, as we somehow acknowledge its existance with the qos settings.
But the moment other rmw implementations create the timestamp may vary.